src/wallet/wallet.cpp

```diff
@@ -2295,33 +2295,38 @@ bool CWalletTx::InMempool() const
 }
 
 bool CWalletTx::IsTrusted(interfaces::Chain::Lock& locked_chain) const
+{
+    std::set<uint256> s;
+    return IsTrusted(locked_chain, s);
+}
+
+bool CWalletTx::IsTrusted(interfaces::Chain::Lock& locked_chain, std::set<uint256>& trusted_parents) const
 {
     // Quick answer in most cases
-    if (!locked_chain.checkFinalTx(*tx)) {
-        return false;
-    }
+    if (!locked_chain.checkFinalTx(*tx)) return false;
     int nDepth = GetDepthInMainChain(locked_chain);
```

Comment:
Maybe I'm missing something but I think you could avoid calling `GetDepthInMainChain` for parents if `nDepth >= 1`?

Reply:
You're right, but we already don't recurse on trusted parents right?

Reply:
Doh!